content/docs/rfcs/72/README.md

```diff
@@ -0,0 +1,273 @@
+---
+slug: 72
+title: 72/WAKU-RLN-KEYSTORE
+name: Waku RLN Keystore
+status: raw
+category: Standards Track
+editor: Jimmy Debe <jimmy@status.im>
+contributors: 
+- ?
+---
+
+# Abstract
+Encrypted credentials are stored in a JSON schema to securely exchange credentials between peers.
+
+# Summary
+A keystore is a construct to store a user’s keys. 
+The keys will be encrypted and decrypted based on methods specified in the specification. 
+This keystore specification uses [32/RLN-V1](https://rfc.vac.dev/spec/32/), Rate Limit Nullifiers, as a spam-prevention mechanism by generating zero-knowledge proofs and storing the proofs locally in the keystore.
+
+# Background
+The secure transfer of keys is important in peer-to-peer messaging applications.
+A Waku RLN Keystore uses zero-knowledge proofs for anonymous rate-limiting for messaging frameworks. 
+Generated credentials by a user are encrypted and stored in the keystore to be retrieved over a network.
+With [32/RLN-V1](https://rfc.vac.dev/spec/32/), sending and receiving messages will ensure a message rate for a network is being followed while keeping the anonymity of the message owner. 
+
+## Example Waku RLN Keystore:
+
+This is an example of a keystore used by a [17/WAKU2-RLN-Relay](https://rfc.vac.dev/spec/17/).
+
+```js
+application: 1 ,
+appIdentifier: 2,
+version: 3,
+credentials: {
+    membershipHash: {
+      crypto: {
+        cipher: 4,
+        cipherparams: 5,
+        ciphertext: 6,
+        kdf: 7,
+        kdfparams: {
+          dklen: 8,
+          c: 9,
+          prf: 10,
+          salt: 11,
+        },
+        mac: 12,
+      }
+    }
+}
+
+```
+
+# Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) and [RFC 8174](https://datatracker.ietf.org/doc/html/rfc8174).
+#
+
+The keystore MUST be generated with a cryptographic construction for password verification and decryption.
+
+Keystore modules MUST include metadata, key derivation function, checksum, cipher, and a membership hash.
+
+
+## Metadata:
+- Information about the keystore SHOULD be stored in the metadata. 
+- The declaration of `application`, `version`, and `appIdentifier` COULD occur in the metadata.
+
+`application` : current application, MUST be a string
+
+`version` : application version, MUST be a string
+
+`appIdentifier`: application identifier, MUST be a string
+
+## Credentials:
+The Waku RLN credentials MUST consist of a `membershipHash` and `WakuCredential`
+- Each contruct MUST include the keypair:
+> key: [membershipHash]: pair: [WakuCredential]
+
+### membershipHash 
+- MUST be a 256 byte hash.
+- it Must be a string
+- SHOULD be generated with `treeIndex`, `membershipContract`, and `identityCredential`.
```

Comment:
please define how it is generated with these values.

Reply:
I made this a little more descriptive, @rymnc